scarb/src/process.rs

```diff
@@ -30,21 +30,58 @@ use crate::core::Config;
 /// itself later exits.
 #[tracing::instrument(level = "debug")]
 pub fn exec_replace(cmd: &mut Command) -> Result<()> {
-    let exit_status = cmd
-        .spawn()
-        .with_context(|| format!("failed to spawn: {}", cmd.get_program().to_string_lossy()))?
-        .wait()
-        .with_context(|| {
-            format!(
-                "failed to wait for process to finish: {}",
-                cmd.get_program().to_string_lossy()
-            )
-        })?;
-
-    if exit_status.success() {
-        Ok(())
-    } else {
-        bail!("process did not exit successfully: {exit_status}");
+    imp::exec_replace(cmd)
+}
+
+#[cfg(unix)]
+mod imp {
+    use anyhow::{bail, Result};
+    use std::os::unix::process::CommandExt;
+    use std::process::Command;
+
+    pub fn exec_replace(cmd: &mut Command) -> Result<()> {
+        let err = cmd.exec();
+        bail!("process did not exit successfully: {err}")
+    }
+}
+
+#[cfg(windows)]
+mod imp {
```

Comment:
Have you tested this on Windows?

Reply:
Doesn't work on Windows, our test hangs... 😰

Reply:
How is it now?

Reply:
It works! C: